### PR TITLE
CIV-12433 Full Defence Not Settle Journey

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-12433

### Change description ###
Update for the full defence not settle journey

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/claimantResponse.ts [hunk 1]
```diff
@@ -104,7 +104,9 @@ export class ClaimantResponse {
   }
 
   hasClaimantAgreedToMediation(): boolean {
-    return this.mediation?.canWeUse?.option === YesNo.YES;
+    return (this.mediation?.canWeUse?.option === YesNo.YES || this.mediation?.canWeUse?.mediationPhoneNumber !== undefined)
```
Comment (paul-pearson) [inline src/main/common/models/claimantResponse.ts:107]: Can use !this.mediation?.canWeUse?.mediationPhoneNumber

File: src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts [hunk 2]
```diff
@@ -26,6 +26,14 @@ function isClaimantWantToSettleTheClaim(claim: Claim) {
   }
 }
 
+function isClaimantWantToProceed(claim: Claim) {
+  if (claim.isFullDefence() && claim.hasPaidInFull()) {
+    return toCCDYesNoReverse(claim.claimantResponse?.hasFullDefenceStatesPaidClaimSettled?.option);
+  } else {
```
[the remark this answers is not in the document]
Comment (paul-pearson) [reply src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts:32]: Could also move this function into Claim class

File: src/main/services/translation/response/convertToCCDYesNo.ts [hunk 3]
```diff
@@ -7,6 +7,12 @@ export const toCCDYesNo = (value: YesNo | string) => {
   }
 };
 
+export const toCCDYesNoReverse = (value: YesNo | string) => {
+  if (value) {
+    return value === YesNo.YES ? YesNoUpperCamelCase.NO : YesNoUpperCamelCase.YES ;
```
Comment (paul-pearson) [inline src/main/services/translation/response/convertToCCDYesNo.ts:12]: Remove space character before ;

File: src/test/unit/services/features/claimantResponse/checkAnswers/buildYourResponseSection.test.ts [hunk 4]
```diff
@@ -117,4 +117,68 @@ describe('Your response Section', () => {
     expect(yourResponseSection.summaryList.rows.length).toBe(4);
     expect(yourResponseSection.summaryList.rows[0].value.html).toContain(t('COMMON.PAYMENT_OPTION.INSTALMENTS', { lng }));
   });
+
+  it('should return Settle The Claim Section when it is Full Defence and Paid with Yes', async () => {
+    //Given
+    const claim = new Claim();
```
Comment (paul-pearson) [inline src/test/unit/services/features/claimantResponse/checkAnswers/buildYourResponseSection.test.ts:123]: As above

File: src/main/common/models/claimantResponse.ts (hunk 1 above)
Comment (kenneth-hmcts) [reply src/main/common/models/claimantResponse.ts]: Need to change to !!this.mediation?.canWeUse?.mediationPhoneNumber instead

File: src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/main/services/translation/claimantResponse/claimantResponseCCDTranslation.ts]: Changed from review and moved to Claim class

File: src/main/services/translation/convertToCUI/cuiTranslation.ts [hunk 5]
```diff
@@ -67,6 +69,7 @@ export const translateCCDCaseDataToCUIModel = (ccdClaimObj: CCDClaim): Claim =>
   claim.interest = claim?.interest ? claim?.interest : translateCCDInterestDetailsToCUI(ccdClaim);
   claim.claimantResponse.suggestedPaymentIntention.paymentOption = toCUIClaimantPaymentOption(ccdClaim.applicant1RepaymentOptionForDefendantSpec);
   claim.claimantResponse.suggestedPaymentIntention.paymentDate = translateCCDPaymentDateToCUIccd(ccdClaim.applicant1RequestedPaymentDateForDefendantSpec?.paymentSetDate);
+  claim.claimantResponse.mediation = ccdClaim.applicant1ClaimMediationSpecRequiredLip ? toCUIClaimantMediation(ccdClaim.applicant1ClaimMediationSpecRequiredLip) : undefined;
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/main/services/translation/convertToCUI/cuiTranslation.ts]: changed from review

File: src/main/services/translation/convertToCUI/cuiTranslation.ts [hunk 6]
```diff
@@ -100,6 +103,12 @@ function translatePartialAdmission(claim: Claim, ccdClaim: CCDClaim, claimantRes
 
 function translateFullDefence(ccdClaim: CCDClaim, claimantResponse: ClaimantResponse): void {
   claimantResponse.intentionToProceed = toCUIGenericYesNo(ccdClaim.applicant1ProceedWithClaim);
+
+  if (ccdClaim?.defenceRouteRequired == CCDRejectAllOfClaimType.HAS_PAID_THE_AMOUNT_CLAIMED
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/main/services/translation/convertToCUI/cuiTranslation.ts]: Changed from review

File: src/main/services/translation/response/convertToCCDYesNo.ts (hunk 3 above)
Comment (kenneth-hmcts) [reply src/main/services/translation/response/convertToCCDYesNo.ts]: Changed from review

File: src/test/unit/common/models/claim.test.ts [hunk 7]
```diff
@@ -2098,4 +2098,75 @@ describe('Documents', () => {
       expect(result).toEqual(true);
     });
   });
+
+  describe('Claim isRejectionReasonCompleted', () => {
+    it('should return false if no claimantResponse object', () => {
+      //Given
+      const claim = new Claim();
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/test/unit/common/models/claim.test.ts]: Changed from review

File: src/test/unit/services/features/claimantResponse/checkAnswers/buildYourResponseSection.test.ts (hunk 4 above)
Comment (kenneth-hmcts) [reply src/test/unit/services/features/claimantResponse/checkAnswers/buildYourResponseSection.test.ts]: Changed from review

File: src/test/unit/common/models/claimantResponse.test.ts [hunk 8]
```diff
@@ -188,4 +190,143 @@ describe('ClaimantResponse model', () => {
     });
 
   });
+
+  describe('hasClaimantNotAgreedToMediation', () => {
+    const claimantResponse = new ClaimantResponse();
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/test/unit/common/models/claimantResponse.test.ts]: Changed from review

File: src/test/unit/common/models/claim.test.ts [hunk 9]
```diff
@@ -2098,4 +2098,75 @@ describe('Documents', () => {
...
+      //When
+      const result = claim.isRejectionReasonCompleted();
+      //Then
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if no hasPartPaymentBeenAccepted object', () => {
+      //Given
+      const claim = new Claim();
+      claim.claimantResponse = new ClaimantResponse();
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/test/unit/common/models/claim.test.ts]: Changed from review

**Review by jeswanth-hmcts — APPROVED** [review]

**Review by paul-pearson — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Pull request merged**